src/core/worker.js

```diff
@@ -436,6 +437,44 @@ var WorkerMessageHandler = PDFJS.WorkerMessageHandler = {
       });
     });
 
+    handler.on('GetRawObject', function wphGetRawObject(data) {
+      function clone(obj) {
+        if (!obj || isBool(obj) || isNum(obj) || isString(obj)) {
+          return obj;
+        } else if (isName(obj)) {
+          return { name: obj.name };
+        } else if (isDict(obj) || isArray(obj)) {
+          return obj.map(clone);
+        } else if (isStream(obj)) {
+          if (obj.str) {
+            return clone(obj.str);
+          } else if (obj.stream) {
+            return clone(obj.stream);
+          } else if (obj.streams) {
+            return clone(obj.streams);
+          }
+          return { dict: clone(obj.dict), start: obj.start, end: obj.end };
+        } else if (isRef(obj)) {
+          return { num: obj.num, gen: obj.gen };
+        }
+        throw new Error('Unknown object type.');
+      }
+
+      if (data.ref === null) {
+        return clone(pdfManager.pdfDocument.xref.trailer);
```

Comment:
I believe that you need to return a promise here, so change it to e.g. this instead:
```javascript
return Promise.resolve(clone(pdfManager.pdfDocument.xref.trailer));
```